ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml

```diff
@@ -291,7 +286,63 @@
+                                            sling:resourceType="granite/ui/components/coral/foundation/form/select"
+                                            fieldLabel="Type"
+                                            name="./displayPatternType">
+                                        <datasource
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="core/fd/components/form/container/v1/datasource"
+                                                fieldType="text-input"
+                                                type="formatters"/>
+                                    </displayPatternType>
+                                    <displayPictureClause
+                                            jcr:primaryType="nt:unstructured"
+                                            granite:class="cmp-adaptiveform-textinput__displayformat"
+                                            sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                            fieldLabel="Format"
+                                            name="./displayFormat">
+                                    </displayPictureClause>
+                                </items>
+                            </displayPatternGroup>
+                            <editPatternGroup
+                                jcr:primaryType="nt:unstructured"
+                                jcr:title="Edit Format"
+                                sling:resourceType="fd/af/authoring/components/granite/form/fieldset"
+                                title="Edit Format">
+                                <items jcr:primaryType="nt:unstructured">
+                                    <editPatternType
+                                            jcr:primaryType="nt:unstructured"
+                                            granite:class="cmp-adaptiveform-textinput__editpattern"
+                                            sling:resourceType="granite/ui/components/coral/foundation/form/select"
+                                            fieldLabel="Type"
+                                            name="./editPatternType">
+                                        <datasource
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="core/fd/components/form/container/v1/datasource"
+                                                fieldType="text-input"
+                                                type="formatters"/>
+                                    </editPatternType>
+                                    <editFormat
+                                            jcr:primaryType="nt:unstructured"
+                                            granite:class="cmp-adaptiveform-textinput__editformat"
```

Comment:
what are the options being listed here for text?

Reply:
Hi @nitin, this works as it used to work in af1. On the basis of the option selected in Patterntype dropdown the corresponding regex gets filled in this editFormat textField. If the author makes some changes in editFormat then the patternType dropdown changes to "Custom".

Reply:
the display and edit patterns should not be listed for text-input.